### ISSUE TITLE
Improve accessibility of geo index map feature inspection panel

### ISSUE THREAD

**Issue by thatbudakguy** [sul-dlss/sul-embed#1792]
@dbranchini noted two issues that SiteImprove flagged when viewing an index map.

- If there is a preview thumbnail for the component map you select, the image doesn't have an `alt`.
- The link to view the component map is a bare URL link, without an accessible name or `title`.

(image below taken from https://purl.stanford.edu/ts545zc6250)

![Screenshot 2023-11-20 at 15 05 40](https://github.com/sul-dlss/sul-embed/assets/4924494/489ab8db-fccd-4ef5-a2d6-0a13773421f7)

The rendering of that sidebar is controlled by [a handlebars template](https://github.com/sul-dlss/sul-embed/blob/main/app/javascript/src/templates/index_map_info.hbs) rendered by the geo viewer: https://github.com/sul-dlss/sul-embed/blob/b871bd74769dec69c5e7115b3b2d5a312ddab5d2/app/javascript/src/modules/geo_viewer.js#L102-L105

The actual content used comes from the `index.json` file that is uploaded as part of the object, which is geoJSON:
```json
{
"type": "FeatureCollection",
"crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:OGC:1.3:CRS84" } },
"features": [
{ "type": "Feature", "properties": { "id": "0", "sheet_id": "G7900 S50 .J3 BHH-8", "call_short": "G7900 S50 .J3", "symph_bar": 36105216323500.0, "call_num": "G7900 S50 .J3 BHH-8", "titleEng": "Chosen 1:50,000 Chikeizu", "objectIdentifier": "qc800wv3208", "websiteUrl": "http:\/\/purl.stanford.edu\/qc800wv3208", "title": "[Chosen 1:50,000 Chikeizu]", "recordIdentifier": "9585363", "sheet": 1, "bbox": "(4:124.5,38.00000000000004,124.75,38.1666666666667)", "webmapUrl": "http:\/\/stanford.maps.arcgis.com\/apps\/SimpleViewer\/index.html?appid=ab4fca3d9d4145e2a5e2fa25deab88b1", "available": "true", "iiifUrl": "https:\/\/purl.stanford.edu\/qc800wv3208\/iiif\/manifest", "label": "BHH-8" }, "geometry": { "type": "MultiPolygon", "coordinates": [ [ [ [ 124.500000000000114, 38.0 ], [ 124.500000000000114, 38.1666666666667 ], [ 124.750000000000114, 38.1666666666667 ], [ 124.750000000000114, 38.0 ], [ 124.500000000000114, 38.0 ] ] ] ] } },
{ "type": "Feature", "properties": { "id": "0", "sheet_id": "G7900 S50 .J3 BHH-4", "call_short": "G7900 S50 .J3", "symph_bar": 36105216323500.0, "call_num": "G7900 S50 .J3 BHH-4", "titleEng": "Chosen 1:50,000 Chikeizu", "objectIdentifier": "ww737hs5709", "websiteUrl": "http:\/\/purl.stanford.edu\/ww737hs5709", "title": "[Chosen 1:50,000 Chikeizu]", "recordIdentifier": "9585363", "sheet": 1, "bbox": "(4:124.75,38.00000000000004,125.0000000000001,38.1666666666667)", "webmapUrl": "http:\/\/stanford.maps.arcgis.com\/apps\/SimpleViewer\/index.html?appid=ab4fca3d9d4145e2a5e2fa25deab88b1", "available": "true", "iiifUrl": "https:\/\/purl.stanford.edu\/ww737hs5709\/iiif\/manifest", "label": "BHH-4" }, "geometry": { "type": "MultiPolygon", "coordinates": [ [ [ [ 124.750000000000114, 38.0 ], [ 124.750000000000114, 38.1666666666667 ], [ 125.000000000000128, 38.1666666666667 ], [ 125.000000000000128, 38.0 ], [ 124.750000000000114, 38.0 ] ] ] ] } },
]
}
```

**Comment by thatbudakguy** [comment]
Given that all we have in terms of data is what's in the `index.json` file, it doesn't seem like we could improve things much. I think the thumbnail image could be considered decorative; the best we could do is just make the alt the title...which is already in the page. I'd propose just setting an empty `alt`.

The website link is a similar conundrum. Maybe we could give it an `aria-title` of "view [map title] online" or something? We don't really have anything to go on except the link text itself. Maybe "view [map title] at [domain name]"?

**Comment by andrewjbtw** [comment]
I noticed that the links to individual maps aren't really active for me. The mouse pointer changes to an icon that suggests the link should be active but nothing happens when I left click. Right-clicking does work. I checked on Firefox and Chrome on MacOS.

For the link title, either of your proposed labels seem like improvements to me. It actually seems a little odd to have the visible header say "Website" instead of something like "Map URL" or "Persistent link" or something more descriptive.

**Comment by dbranchini** [comment]
I agree that the label, "website," isn't as descriptive as it could be. Oddly, in addition to the image link, the website link doesn't work for me either.  

I'd suggest that the image should not be a link, if the link below it goes to the same place. See https://usability.yale.edu/web-accessibility/articles/links#image-links for reference. It's best practice to avoid two separate links adjacent to each other that go to the same place.

**Comment by thatbudakguy** [comment]
It looks like the links aren't working because they are `http://` instead of `https://` which triggers a mixed content error because the content was loaded within an iframe, so the browser will block navigation. This is a build-in security feature of many browsers.

Those links are actually in the `index.json` metadata, so there's probably some remediation needed to fix them. I don't know how those files are generated, but maybe we need to change code somewhere else too. We shouldn't ever need to link to purl non-securely (i.e. with `http://` only).

In the meantime, I think we should:

- [ ] give the thumbnail image an empty alt, indicating it is decorative, since we can't describe it
- [ ] remove the link around the thumbnail image, since we don't need two links
- [ ] change the bare link text to "View this map" instead of just the purl URL
- [ ] remove the "website" heading since it's redundant with the words of the link itself
